osu.Game/Screens/OnlinePlay/DailyChallenge/DailyChallengeIntro.cs

```diff
@@ -0,0 +1,409 @@
+            }, true);
+
+            LoadComponentAsync(new OnlineBeatmapSetCover(item.Beatmap.BeatmapSet as IBeatmapSetOnlineInfo)
+            {
+                RelativeSizeAxes = Axes.Both,
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                FillMode = FillMode.Fit,
+                Scale = new Vector2(1.2f),
+                Shear = new Vector2(-OsuGame.SHEAR, 0f),
+            }, c =>
+            {
+                beatmapBackground.Add(c);
+
+                beatmapBackgroundLoaded = true;
+                updateAnimationState();
+            });
+        }
+
+        protected override void Update()
+        {
+            base.Update();
+
+            if (trackContent)
+            {
+                float vShift = initial_v_shift + (beatmapContent.DrawHeight * beatmapContent.Scale.Y) / 2;
+
+                float yPos = (float)Interpolation.DampContinuously(bottomDateDisplay.Y, vShift, 16, Clock.ElapsedFrameTime);
+                float xPos = (float)Interpolation.DampContinuously(bottomDateDisplay.X, getShearForY(vShift) + final_v_shift, 16, Clock.ElapsedFrameTime);
+
+                topTitleDisplay.Position = new Vector2(-xPos, -yPos);
+                bottomDateDisplay.Position = new Vector2(xPos, yPos);
+            }
+        }
+
+        public override void OnEntering(ScreenTransitionEvent e)
+        {
+            base.OnEntering(e);
+            this.FadeInFromZero(400, Easing.OutQuint);
+            updateAnimationState();
+        }
+
+        public override void OnSuspending(ScreenTransitionEvent e)
+        {
+            this.FadeOut(800, Easing.OutQuint);
+            base.OnSuspending(e);
+        }
+
+        private void updateAnimationState()
+        {
+            if (!beatmapBackgroundLoaded || !this.IsCurrentScreen())
+                return;
+
+            if (animationBegan)
+                return;
+
+            beginAnimation();
+            animationBegan = true;
+        }
+
+        private void beginAnimation()
+        {
+            const float v_spacing = 0;
+
+            using (BeginDelayedSequence(200))
+            {
+                introContent.Show();
+
+                topTitleDisplay.MoveToOffset(new Vector2(getShearForY(-initial_v_shift), -initial_v_shift));
+                bottomDateDisplay.MoveToOffset(new Vector2(getShearForY(initial_v_shift), initial_v_shift));
+
+                topTitleDisplay.MoveToX(getShearForY(topTitleDisplay.Y) - 500)
+                               .MoveToX(getShearForY(topTitleDisplay.Y) - v_spacing, 300, Easing.OutQuint)
+                               .FadeInFromZero(400, Easing.OutQuint);
+
+                bottomDateDisplay.MoveToX(getShearForY(bottomDateDisplay.Y) + 500)
+                                 .MoveToX(getShearForY(bottomDateDisplay.Y) + v_spacing, 300, Easing.OutQuint)
+                                 .FadeInFromZero(400, Easing.OutQuint);
```

Comment:
This doesn't feel quite right to me, I can't really put my finger on it. I think it may be that it doesn't sync with the rest of the display well enough and it just looks like two things flying out the sides.

I'm thinking maybe having the centre part expand horizontally then vertically as these move would look better but that's just theorising.

Reply:
Yeah I'm also not 100% happy with things. I'll probably revisit this in a second pass (or accept any input from others) but hit my time limit for spending time on tweaking things today.

Will check on tests.